src/main/resources/com/google/api/codegen/clientconfig/client_gapic.yaml

```diff
@@ -0,0 +1,6 @@
+type: com.google.api.codegen.ConfigProto
+templates:
+- codegen_provider: com.google.api.codegen.clientconfig.ClientConfigGapicProvider
+  input_element_view: com.google.api.codegen.InterfaceView
+  snippet_files:
+  - main.snip
```

Comment:
I think you might rename main.snip to json.snip, and add an xml.snip. I doubt the Java classes would need to change.